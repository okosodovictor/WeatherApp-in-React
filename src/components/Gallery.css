.photoThumbnail {
  width: 120px !important;
  height: 80px !important;
  margin: 5px !important;
}
.largePhoto {
  max-height: 400px;
  max-width: 600px;
}
