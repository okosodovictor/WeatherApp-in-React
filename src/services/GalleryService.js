function GalleryService() {
  return {
    getPhotos: async function() {
      let response = await fetch("/api/photos");
      return response;
    },
    getImageById: async function(id) {
      let response = await fetch("/api/photos/" + id);
      return response;
    }
  };
}

export default GalleryService;
