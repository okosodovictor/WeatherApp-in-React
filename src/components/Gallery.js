import React, { Component } from "react";
import GalleryService from "../services/GalleryService";
import { Link } from "react-router-dom";

class Gallery extends Component {
  constructor(props) {
    super(props);
    this.state = { photos: [] };
  }

  render() {
    return (
      <div className="container">
        <h2>Photo Gallery</h2>
        <hr />
        <div className="row">
          {this.state.photos.map(photo => (
            <Link to={`/gallery/${photo.id}`}>
              <img
                title={`${photo.title} - ${photo.description}`}
                src={photo.thumbnail}
                className="img img-thumbnail photoThumbnail"
                alt=""
              />
            </Link>
          ))}
        </div>
      </div>
    );
  }

  async componentDidMount() {
    let photos = await GalleryService().getPhotos();
    let response = await photos.json();
    this.setState({ photos: response });
  }
}

export default Gallery;
