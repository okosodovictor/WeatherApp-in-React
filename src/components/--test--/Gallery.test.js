import React from "react";
import Gallery from "../Gallery";
import { shallow } from "../../enzyme";
import GalleryService from "../../services/GalleryService";
import { log } from "util";
describe("Contact tests", () => {
  const myMock = jest.fn();

  beforeEach(function() {
    const photos = [
      {
        id: 1,
        title: "galaxy",
        description: "Galaxy of stars",
        thumbnail: "images/image-1.jpg",
        large: "images/image-1-large.jpg"
      },
      {
        id: 2,
        title: "amazon",
        description: "Tick amazon forest",
        thumbnail: "images/image-2.jpg",
        large: "images/image-2-large.jpg"
      },
      {
        id: 3,
        title: "Cloud",
        description: "Tick white cloud",
        thumbnail: "images/image-3.jpg",
        large: "images/image-3-large.jpg"
      },
      {
        id: 4,
        title: "",
        description: "Ice",
        thumbnail: "images/image-4.jpg",
        large: "images/image-4-large.jpg"
      },
      {
        id: 5,
        title: "Sky",
        description: "Sunny Cloud",
        thumbnail: "images/image-5.jpg",
        large: "images/image-5-large.jpg"
      },
      {
        id: 6,
        title: "Sky2",
        description: "Nice Cloud Image",
        thumbnail: "images/image-6.jpg",
        large: "images/image-6-large.jpg"
      },
      {
        id: 7,
        title: "Sky3",
        description: "another cloud image",
        thumbnail: "images/image-7.jpg",
        large: "images/image-7-large.jpg"
      },
      {
        id: 8,
        title: "sky4",
        description: "sky image from the river bank",
        thumbnail: "images/image-8.jpg",
        large: "images/image-8-large.jpg"
      },
      {
        id: 9,
        title: "sky 5",
        description: "another sky image",
        thumbnail: "images/image-9.jpg",
        large: "images/image-9-large.jpg"
      },
      {
        id: 10,
        title: "sky 6",
        description: "Raining street image",
        thumbnail: "images/image-10.jpg",
        large: "images/image-10-large.jpg"
      }
    ];
    global.fetch = jest.fn().mockImplementation(() => Promise.resolve(photos));
  });
  let gallery;

  it("renders without crashing", () => {
    gallery = shallow(<Gallery />);
  });

  it("get Photos should return 10 photo", async function() {
    const response = await GalleryService().getPhotos();
    expect(response).toHaveLength(10);
  });

  it("calls componentDidMount", () => {
    jest.spyOn(Gallery.prototype, "componentDidMount");
    gallery = shallow(<Gallery />);
    expect(Gallery.prototype.componentDidMount.mock.calls.length).toBe(1);
  });
});
