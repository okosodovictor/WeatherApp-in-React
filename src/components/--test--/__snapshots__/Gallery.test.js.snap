// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact tests get Photos should return 10 photo 1`] = `
Array [
  Object {
    "description": "Galaxy of stars",
    "id": 1,
    "large": "images/image-1-large.jpg",
    "thumbnail": "images/image-1.jpg",
    "title": "galaxy",
  },
  Object {
    "description": "Tick amazon forest",
    "id": 2,
    "large": "images/image-2-large.jpg",
    "thumbnail": "images/image-2.jpg",
    "title": "amazon",
  },
  Object {
    "description": "Tick white cloud",
    "id": 3,
    "large": "images/image-3-large.jpg",
    "thumbnail": "images/image-3.jpg",
    "title": "Cloud",
  },
  Object {
    "description": "Ice",
    "id": 4,
    "large": "images/image-4-large.jpg",
    "thumbnail": "images/image-4.jpg",
    "title": "",
  },
  Object {
    "description": "Sunny Cloud",
    "id": 5,
    "large": "images/image-5-large.jpg",
    "thumbnail": "images/image-5.jpg",
    "title": "Sky",
  },
  Object {
    "description": "Nice Cloud Image",
    "id": 6,
    "large": "images/image-6-large.jpg",
    "thumbnail": "images/image-6.jpg",
    "title": "Sky2",
  },
  Object {
    "description": "another cloud image",
    "id": 7,
    "large": "images/image-7-large.jpg",
    "thumbnail": "images/image-7.jpg",
    "title": "Sky3",
  },
  Object {
    "description": "sky image from the river bank",
    "id": 8,
    "large": "images/image-8-large.jpg",
    "thumbnail": "images/image-8.jpg",
    "title": "sky4",
  },
  Object {
    "description": "another sky image",
    "id": 9,
    "large": "images/image-9-large.jpg",
    "thumbnail": "images/image-9.jpg",
    "title": "sky 5",
  },
  Object {
    "description": "Raining street image",
    "id": 10,
    "large": "images/image-10-large.jpg",
    "thumbnail": "images/image-10.jpg",
    "title": "sky 6",
  },
]
`;
